### PR TITLE
RHPAM-1761 Bulk abort throws an exception when called on several subp…

### PR DESCRIPTION
**Pull request by bxf12315**
…rocess instances

### CODE REVIEW

**Comment by bxf12315** [comment]
@mswiderski I want to test issue RHPAM-1761, when I abort parent process instance  , I excepted all child processes instance  of the parent process instance will close, but I find (https://github.com/kiegroup/jbpm/pull/1406/files#diff-e4fac608fcfd24fde45751c3e3449fd1R311)these instances is active, Please give me some advise.
Thanks

**Comment by mswiderski** [comment]
@bxf12315 your call activity has a independent flag set to true which means that sub process will not be aborted when parent is

**Comment by bxf12315** [comment]
@mswiderski How to set the flag to false ? I need create a unit test for abort parent and child processes.

**Comment by mswiderski** [comment]
there is a drop down in designer to change the independent flag, not sure this id available in both stunner and designer but for sure in designer

**Comment by mswiderski** [comment]
@bxf12315 here is a screen shot from stunner when call activity/reusable subprocess is selected

image

there is Independent checkbox that allows to control that behaviour

**Comment by bxf12315** [comment]
> @bxf12315 why not just simply catch exception of ProcessInstanceNotFound when looping through instances to abort? That way you can just log a warning and continue with other instances to be aborted. No need to do any extra processing or loading data

I agree with you, it is very simply way. I have one concern, client never get  ProcessInstanceNotFound exception.

**Comment by mswiderski** [comment]
correct, ProcessInstanceNotFoundException is never returned to client, simply logged and ignored when doing abort with many instances

**Comment by bxf12315** [comment]
@mswiderski  updated it.

**Comment by cristianonicolai** [comment]
jenkins please retest this

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java [hunk 1]
```diff
@@ -268,6 +270,47 @@ public void testStartAndAbortProcesses() {
     	assertNull(pi2);
     }
 
+    @Test
+	public void testAbortedParentProcessAndChildProcessInProcessInstanceList(){
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java]: no, just test if abort sub and parent processes do not have exception.

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java [hunk 2]
```diff
@@ -190,21 +190,29 @@ public void abortProcessInstance(String deploymentId, Long processInstanceId) {
...
-	
-	@Override
+
+    @Override
     public void abortProcessInstances(List<Long> processInstanceIds) {
-	    for (long processInstanceId : processInstanceIds) {
-            abortProcessInstance(processInstanceId);
-        }
-	}
+        processInstanceIds.stream().forEach(processInstanceId -> {
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java]: Sorry

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java [hunk 3]
```diff
@@ -268,6 +270,47 @@ public void testStartAndAbortProcesses() {
     	assertNull(pi2);
     }
 
+    @Test
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java]: ok

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java [hunk 4]
```diff
@@ -190,21 +190,29 @@ public void abortProcessInstance(String deploymentId, Long processInstanceId) {
...
+            try {
+                abortProcessInstance(processInstanceId);
+            } catch (ProcessInstanceNotFoundException processInstanceNotFoundException) {
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java]: https://github.com/kiegroup/jbpm/blob/11da6eda8b1f964aedb7a89db4333c5d42546897/jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java#L167  
have same exception message.

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java]: no, it is a test that here do not have exception when abort sub and parent processes in a process instance list.

[the remark this answers is not in the document]
Comment (cristianonicolai) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java]: could you add that then?

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java]: I'm fine if it repeats the message, the main point to me is to have some message and the actual exception logged.

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java [hunk 5]
```diff
@@ -268,6 +275,91 @@ public void testStartAndAbortProcesses() {
...
+    private void testResultForAbortedParentProcessAndChildProcessInProcessInstanceList(Long parentProcessInstanceId){
+        Collection<ProcessInstanceDesc> instances = runtimeDataService.getProcessInstancesByParent(parentProcessInstanceId, Arrays.asList(ProcessInstance.STATE_ACTIVE), new QueryContext());
+        assertNotNull(instances);
+        assertEquals(2, instances.size());
+
+        List<Long> processInstanceList = instances.stream().map(p -> p.getId()).collect(Collectors.toList());
+        processInstanceList.add(0, parentProcessInstanceId);
+
+        processService.abortProcessInstances(processInstanceList);
+        processService.abortProcessInstances(processInstanceList);
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java]: sorry

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java [hunk 6]
```diff
@@ -120,6 +125,8 @@ public void cleanup() {
                 deploymentService.undeploy(unit);
             	} catch (Exception e) {
             		// do nothing in case of some failed tests to avoid next test to fail as well
+                    logger.error("---------------------------" + e.getLocalizedMessage());
+                    logger.error("--do nothing in case of some failed tests to avoid next test to fail as well-------------------------" + unit.getIdentifier());
```
Comment (MarianMacik) [inline jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java]: Do we need this logging? Maybe we can at least decrease the logging level to debug so it doesn't pollute logs when tests are run.

Comment (bxf12315) [reply jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplTest.java]: My debug code, I removed it.

**Pull request merged**